server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -8,28 +8,349 @@
+        orchestrator.setTransportService(transportService);
+
+        try (MockLogAppender mockLogAppender = MockLogAppender.createForLoggers(LogManager.getLogger(ExtensionsOrchestrator.class))) {
+
+            mockLogAppender.addExpectation(
+                new MockLogAppender.SeenEventExpectation(
+                    "Transport Null Exception 1",
+                    "org.opensearch.extensions.ExtensionsOrchestrator",
+                    Level.ERROR,
+                    "ConnectTransportException[[][127.0.0.0:9300] connect_timeout[30s]]"
+                )
+            );
+
+            mockLogAppender.addExpectation(
+                new MockLogAppender.SeenEventExpectation(
+                    "Transport Connect Exception 2",
+                    "org.opensearch.extensions.ExtensionsOrchestrator",
+                    Level.ERROR,
+                    "ConnectTransportException[[][127.0.0.1:9301] connect_exception]; nested: ConnectException[Connection refused];"
+                )
+            );
+
+            orchestrator.extensionsInitialize();
+
+            transportService.close();
+            ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+            mockLogAppender.assertAllExpectationsMatched();
+        }
+    }
+
+    public void testOnIndexModule() throws Exception {
+
+        Path extensionsDir = createTempDir().resolve("extensions");
+        Path pluginDir1 = extensionsDir.resolve("fake-extension-1");
+        Path pluginDir2 = extensionsDir.resolve("fake-extension-2");
+
+        PluginTestUtil.writePluginProperties(
+            pluginDir1,
+            "description",
+            "fake desc",
+            "name",
+            "firstExtension",
+            "version",
+            "1.0",
+            "opensearch.version",
+            Version.CURRENT.toString(),
+            "java.version",
+            System.getProperty("java.specification.version"),
+            "classname",
+            "FakeExtensions"
+        );
+
+        PluginTestUtil.writePluginProperties(
+            pluginDir2,
+            "description",
+            "fake desc",
+            "name",
+            "secondExtension",
+            "version",
+            "1.0",
+            "opensearch.version",
+            Version.CURRENT.toString(),
+            "java.version",
+            System.getProperty("java.specification.version"),
+            "classname",
+            "FakeExtensions"
+        );
+
+        Settings settings = Settings.builder()
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .build();
+        ExtensionsOrchestrator orchestrator = new ExtensionsOrchestrator(settings, extensionsDir);
+
+        ThreadPool threadPool = new TestThreadPool(ExtensionsOrchestratorTests.class.getSimpleName());
+        MockNioTransport transport = new MockNioTransport(
```

Comment:
Here also, mock Netty instead of Nio

Reply:
I haven't been able to find a Mock Netty file.  All of the plugin tests use this form of transport instead.